site/docs/row-level-deletes.md

```diff
@@ -0,0 +1,196 @@
+However, because we can quickly know the row position of unflushed data, it is more desirable to write a position delete for it.
+
+## Compaction
+
+In Iceberg, we are interested in the following 2 types of compactions:
+
+- **data compaction**: optimization of data files into better layout through algorithms like bin-packing, sorting, z-ordering, etc.
+- **delete compaction**: optimization of delete files into better layout, which also includes removing delete files
+
+!!!Note
+    The big data community commonly uses the terms _major compaction_ and _minor compaction_ to describe compaction features. 
+    However, we notice that using these terms could easily cause confusion because people have different definitions for these terms.
+    Therefore, we do not use these terms in this article, and we will focus on explaining specific Iceberg actions that compact data and delete files.
+
+### Data compaction
+
+Iceberg data compaction is done through the `RewriteDataFiles` action.
+Users can choose different rewrite strategies such as bin-packing or sorting to rewrite files into optimized layout.
+More details could be found in the procedure pages for each engine that implements this action, including Spark and Flink.
+
+### Delete compaction
+
+Accumulating too many deletes would result in high latency at read time.
+Having too many equality deletes might even cause out-of-memory error.
+Delete compaction should be used to keep the number of delete files low in an Iceberg table.
+
+The most straight-forward way to achieve delete compaction is to merge deletes into data files and produce new data files that do not contain the removed rows.
+Notice that data compaction already merges deletes as a part of reading data files, so 2 configurations are added to `RewriteDataFiles` to also support delete compaction:
+
+| Property                          | Default             | Description                                            |
+| --------------------------------- | ------------------- | ------------------------------------------------------ |
+| delete-file-threshold             | `Integer.MAX_VALUE` | The minimum number of deletes that needs to be associated with a data file for it to be considered for rewriting. If a data file has this number of deletes or more, it will be rewritten regardless of its file size. |
+| use-starting-sequence-number      | true                | If the compaction should use the sequence number of the snapshot at compaction start time for new data files, instead of using the sequence number of the newly produced snapshot. The use case was discussed in section [position delete vs equality delete](#position-delete-vs-equality-delete) to minimize commit conflicts. |
+
+#### Additional compaction actions
```

Comment:
This is good documentation which helps users to understand the design of copy on write and merge on read deletes. 

However many people are confused and interested to know about which engine supports copy on write and which engine supports merge on read deletes with iceberg?
**So, can we have a table for that in the additional section here ?**
Also If single engine supports both type of deletes, then we should mention how to switch between the type of deletes.

Reply:
I think a compatibility matrix is likely needed for each engine as we are also introducing multi-version support. So I am not including it here.